src/transformers/models/pix2struct/modeling_pix2struct.py

```diff
@@ -0,0 +1,1698 @@
+        r"decoder.layer.0.layer.1.EncDecAttention.relative_attention_bias.weight",
+    ]
+
+    def __init__(self, config: Pix2StructConfig):
+        super().__init__(config)
+        encoder_config = copy.deepcopy(config.vision_config)
+        self.encoder = Pix2StructVisionModel(encoder_config)
+
+        decoder_config = copy.deepcopy(config.text_config)
+        self.decoder_input_ids = decoder_config.pad_token_id
+        self.decoder_eos_token_ids = decoder_config.eos_token_id
+        self.decoder = Pix2StructTextModel(decoder_config)
+
+        self.is_vqa = config.is_vqa
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def get_input_embeddings(self):
+        return self.shared
+
+    def set_input_embeddings(self, new_embeddings):
+        self.shared = new_embeddings
+        self.decoder.set_input_embeddings(new_embeddings)
+
+    def get_decoder(self):
+        return self.decoder
+
+    def get_encoder(self):
+        return self.encoder
+
+    @add_start_docstrings_to_model_forward(Pix2StructText_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=Seq2SeqModelOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        pixel_embeds: Optional[torch.FloatTensor] = None,
+        attention_mask: Optional[torch.FloatTensor] = None,
+        decoder_input_ids: Optional[torch.LongTensor] = None,
+        decoder_attention_mask: Optional[torch.BoolTensor] = None,
+        head_mask: Optional[torch.FloatTensor] = None,
+        decoder_head_mask: Optional[torch.FloatTensor] = None,
+        cross_attn_head_mask: Optional[torch.Tensor] = None,
+        encoder_outputs: Optional[Tuple[Tuple[torch.FloatTensor]]] = None,
+        past_key_values: Optional[Tuple[Tuple[torch.FloatTensor]]] = None,
+        labels: Optional[torch.LongTensor] = None,
+        decoder_inputs_embeds: Optional[torch.Tensor] = None,
+        use_cache: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple[torch.FloatTensor], Seq2SeqModelOutput]:
+        r"""
+        Returns:
+
+        Example:
+
+        ```python
+        >>> from transformers import AutoTokenizer, Pix2StructTextModel
+
+        >>> tokenizer = AutoTokenizer.from_pretrained("t5-small")
+        >>> model = Pix2StructTextModel.from_pretrained("t5-small")
+
+        >>> input_ids = tokenizer(
+        ...     "Studies have been shown that owning a dog is good for you", return_tensors="pt"
+        ... ).input_ids  # Batch size 1
+        >>> decoder_input_ids = tokenizer("Studies show that", return_tensors="pt").input_ids  # Batch size 1
+
+        >>> # preprocess: Prepend decoder_input_ids with start token which is pad token for Pix2StructTextModel.
+        >>> # This is not needed for torch's Pix2StructTextForConditionalGeneration as it does this internally using labels arg.
```

Comment:
Should this doc example be used for `Pix2StructTextModel` and a different one used for `Pix2StructTextForConditionalGeneration` ?